### PR TITLE
feat: warn about uppercase letters when ignoring case globally

### LINKED ISSUE #113: Check for uppercase characters when using the `i` flag
When the `i` flag is used, the entire regular expression will be case insensitive and all upper case letters can be replaced by their lowercase counterparts.

### PR DESCRIPTION
**Pull request by fzipi**
Fixes #113

### CODE REVIEW

**Comment by theseion** [comment]
I like it but I'd rather have it in `regex_format.go`, so it will fail the check on GH when we run the format checker.

**Comment by fzipi360** [comment]
Moved!

**Comment by fzipi** [comment]
Moved!

**Review by theseion — CHANGES_REQUESTED** [review]

File: cmd/regex_format.go [hunk 1]
```diff
@@ -164,13 +167,33 @@ func processFile(filePath string, ctxt *processors.Context, checkOnly bool) erro
...
+			lastByte := len(parsedBytes.Bytes())
+			if showLast > lastByte {
+				showLast = lastByte
+			}
+			if showFirst < 0 {
+				showFirst = 0
+			}
+			logger.Warn().Msgf("File contains uppercase letters, but ignore-case flag is set. Please check your source files.")
+			logger.Warn().Msgf("Problem found around char %d:\n%s\n", found, parsedBytes.String()[showFirst:showLast])
+			logger.Warn().Msg("Be aware that because of file inclusions and definitions, the actual line number or file might be different.")
```
[the remark this answers is not in the document]
Comment (fzipi) [reply cmd/regex_format.go:193]: Sure, changed.

File: cmd/regex_format.go [hunk 2]
```diff
@@ -164,13 +167,39 @@ func processFile(filePath string, ctxt *processors.Context, checkOnly bool) erro
...
+			message := "File not properly formatted"
```
Comment (theseion) [inline cmd/regex_format.go:194]: I would only print the message and return the value if `checkOnly` is true to be consistent with the behaviour below.

**Review by theseion — CHANGES_REQUESTED** [review]

File: cmd/regex_format.go (hunk 2 above)
Comment (fzipi) [reply cmd/regex_format.go]: Sure. Did reorganize things a bit to DRY.

**Review by theseion — APPROVED** [review]
Awesome!

**Pull request merged**